src/main/java/org/jabref/gui/preferences/externalfiletypes/ExternalFileTypesTabViewModel.java

```diff
@@ -82,7 +82,12 @@ public boolean edit(ExternalFileTypeItemViewModel type) {
         ExternalFileTypeItemViewModel typeToModify = new ExternalFileTypeItemViewModel(type.toExternalFileType());
         showEditDialog(typeToModify, Localization.lang("Edit file type"));
 
-        if (!isValidExternalFileType(typeToModify)) {
+        if (type.extensionProperty().get().equals(typeToModify.extensionProperty().get())) {
```

Comment:
I have no clue why this works. Can you add a Java comment?

In my understanding, this evaluates to true if the extension is the same (either empty or set). Then the check `withEmptyValue` is done.

In the "old" branch, the `withEmptyValue` is also done (indirectly via `isValidExternalFileType`.

I really don't get why this fixes the issue ^^.

Reply:
The isValidExternalType checks if the file extension is already used before. This will always return true in the case of **edit** and return false -> thus you can never edit one file typ entry. 
Therefore, we check if we still have the same extension -> we just edited an existing one by e.g. changing the application path